packages/kbn-rule-data-utils/src/technical_field_names.ts

```diff
@@ -97,6 +97,14 @@ const ALERT_RULE_CONSUMER = `${ALERT_RULE_NAMESPACE}.consumer` as const;
 const ALERT_RULE_PRODUCER = `${ALERT_RULE_NAMESPACE}.producer` as const;
 const ALERT_RULE_UUID = `${ALERT_RULE_NAMESPACE}.uuid` as const;
 
+// Fields related to Alert context
+const ALERT_CONTEXT_CONTAINER = 'container' as const;
```

Comment:
Are these all ECS fields? If they are, would you consider just adding the ECS component template to the index template for these AAD indices instead of defining them separately in the technical field mappings? This way their mapping does not diverge from the official ECS mapping.

Reply:
These are ECS fields. The ECS component template is already added to index template with -
```componentTemplateRefs: [ECS_COMPONENT_TEMPLATE_NAME],``` for `infra` rules.
And the `ecsFieldMap` has flattened fields (`cloud.instance.id`, `cloud.instance.name`, etc.), but not root level fields like `cloud.*` that I need here.
What would be the best approach in this scenario?

Reply:
Sorry if I'm not understanding. Are you saying the ECS field map contains `cloud.instance.id` but you might be passing in `{cloud: { instance: {id: ''}}}`? Could you flatten the keys? That would be consistent with the other keys in the AAD indices.

Reply:
Also, we recently merged [this PR](https://github.com/elastic/kibana/pull/143703) so I think unflattening the field for the action template may no longer be necessary (please correct me if I'm wrong!)

Reply:
By action template, I mean in the UI. To avoid having too many fields, we are only displaying top level fields. I understand your concern. Let me have a look at the PR and see if we can work with keeping fields as is in action template.

![image](https://user-images.githubusercontent.com/69037875/200846268-53654e77-ff64-4e88-97b1-c224a2b21a5d.png)

Reply:
I took a look at the index mapping generated by the field template and it looks like even though the field mapping defines the keys as flattened `cloud.account.id`, the index mapping is not flattened

```
"cloud": {
        "properties": {
          "account": {
            "properties": {
              "id": {
                "type": "keyword"
              },
          },
}
```

This is why when you initially added the fields to the AAD documents in https://github.com/elastic/kibana/pull/143001, they show up and are mapped (I was wondering about this since dynamic mapping is set to false for the index).

Context variables are defined separately from the AAD mappings so I think you should be able to define a context variable of `cloud` without having to specify all the subkeys. We have another rule type that returns ES documents inside a `context.hits` variable and we don't need to define exactly what the hits look like.

Reply:
thank you @ymao1 for your input, I am working on the changes.